gazelle/buf/buf_lock.go

```diff
@@ -0,0 +1,22 @@
+package buf
+
+import "os"
+
+type bufLock struct {
```

Comment:
I think we should have some comment to what code to bufbuild/buf this should be kept in sync to (assuming sharing structs with bufbuild/buf is out of the question).